packages/opencensus-exporter-stackdriver/src/stackdriver-cloudtrace.ts

```diff
@@ -70,51 +72,66 @@ export class StackdriverTraceExporter implements Exporter {
-   */
-  private translateTrace(root: RootSpan): TranslatedTrace {
-    const spanList = root.spans.map((span: Span) => this.translateSpan(span));
-    spanList.push(this.translateSpan(root));
-
-    return {projectId: this.projectId, traceId: root.traceId, spans: spanList};
+  async translateSpan(rootSpans: RootSpan[]) {
+    const resourceLabel = await this.RESOURCE_LABELS;
+    const spanList: Span[] = [];
+    rootSpans.forEach(rootSpan => {
+      // RootSpan data
+      spanList.push(this.createSpan(rootSpan, resourceLabel));
+      rootSpan.spans.forEach(span => {
+        // Builds spans data
+        spanList.push(this.createSpan(span, resourceLabel));
+      });
+    });
+    return spanList;
   }
 
-  /**
-   * Translates span data to Stackdriver's span format.
-   * @param span
-   */
-  private translateSpan(span: Span): TranslatedSpan {
-    return {
-      name: span.name,
-      kind: 'SPAN_KIND_UNSPECIFIED',
-      spanId: hexToDec(span.id),
-      startTime: span.startTime,
-      endTime: span.endTime,
-      labels: Object.keys(span.attributes)
-                  .reduce(
-                      (acc, k) => {
-                        acc[k] = String(span.attributes[k]);
-                        return acc;
-                      },
-                      {} as Record<string, string>)
+  private createSpan(
+      span: OCSpan, resourceLabels: Record<string, AttributeValue>): Span {
+    const spanName =
+        `projects/${this.projectId}/traces/${span.traceId}/spans/${span.id}`;
+
+    const spanBuilder: Span = {
+      name: spanName,
+      spanId: span.id,
+      displayName: stringToTruncatableString(span.name),
+      startTime: span.startTime.toISOString(),
+      endTime: span.endTime.toISOString(),
+      attributes: createAttributes(
+          span.attributes, resourceLabels, span.droppedAttributesCount),
+      timeEvents: createTimeEvents(
+          span.annotations, span.messageEvents, span.droppedAnnotationsCount,
+          span.droppedMessageEventsCount),
+      links: createLinks(span.links, span.droppedLinksCount),
+      status: {code: span.status.code},
+      sameProcessAsParentSpan: !span.remoteParent,
+      childSpanCount: null,
+      stackTrace: null,  // Unsupported by nodejs
     };
+    if (span.parentSpanId) {
+      spanBuilder.parentSpanId = span.parentSpanId;
+    }
+    if (span.status.message) {
+      spanBuilder.status.message = span.status.message;
+    }
+
+    return spanBuilder;
   }
 
   /**
-   * Sends traces in the Stackdriver format to the service.
-   * @param traces
+   * Sends new spans to new or existing traces in the Stackdriver format to the
+   * service.
+   * @param spans
    */
-  private sendTrace(traces: TracesWithCredentials) {
+  private batchWriteSpans(spans: SpansWithCredentials) {
     return new Promise((resolve, reject) => {
-      cloudTrace.projects.patchTraces(traces, (err: Error) => {
+      cloudTrace.projects.traces.batchWrite(spans, (err: Error) => {
```

Comment:
Any reason we are using HTTP and not gRPC? (If the answer is just "that's what we did before", could you create an issue to track moving to gRPC for this?)

Reply:
Good point, added TODO here.